SPIKE.md

```diff
@@ -0,0 +1,92 @@
+# Spike: Robust handling of ObjectID type for MongoDB
+
+Ref: https://github.com/strongloop/loopback-next/issues/3456
+
+## Table of contents
+
+- [The problem](#the-problem)
+- [Proposed solution](#proposed-solution)
+- [Approaches](#approaches)
+- [Developer Notes](#developer-notes)
+- [Implementation Proposal](#implementation)
+- [Follow-up task](#follow-up-task)
+
+## The Problem
+
+`ObjectID` is a native MongoDB datatype, which is used as the datatype of its primary key (_id_) field; other fields
+can also be of `ObjectID` type.
+
```

Comment:
Let's add a short paragraph to explain that:

MongoDB distinguishes between id values that are a `string` and an `ObjectID`. A query using a `string` value does not match records having `ObjectID` value and vice versa. This is especially important for foreign keys, e.g. when `Order` property `customerId` is referencing an `id` of a `Customer`, then both properties must always hold `ObjectID` values. If one of the property (typically `customerId`) is stored as a `string`, then LB relations stop working.